engine-precompiles/src/native.rs

```diff
@@ -791,6 +787,30 @@ fn parse_input(input: &[u8]) -> Result<&[u8], ExitError> {
     Ok(&input[1..])
 }
 
+#[derive(serde::Serialize, serde::Deserialize)]
+struct FtTransferCallArgs {
```

Comment:
Performance nit: we could avoid some cloning in the `ft_transfer_call_args` implementation by holding references to `receiver_id` and `msg` in this struct and not deriving `Deserialize`.